README.md
<h1 align="center">
    <br>
    <img width="328" src="https://user-images.githubusercontent.com/12554487/100519844-68254480-31d5-11eb-9834-f926818d1aad.png" alt="webapp-suite ui logo">
    <br>
</h1>

<p align="center">
  Simple style react components for modern web apps
    <a href="https://ui.webapp-suite.com">
      https://ui.webapp-suite.com
    </a>
</p>

<div align="center">

[![Build Status][travis-image]][travis-url]
[![Codecov][codecov-image]][codecov-url]
[![Npm Version][npm-version-image]][npm-version-url]
[![Downloads Per Month][npm-downloads-image]][npm-downloads-url]
[![Dependencies][dependencies-image]][dependencies-url]
[![Open PRs][open-prs-image]][open-prs-url]
[![Contributors][contributors-image]][contributors-url]
[![Slack][slack-image]][slack-url]
[![License][license-image]][license-url]
[![License Scan][license-scan-image]][license-scan-url]
[![Last Commit][last-commit-image]][last-commit-url]

</div>

## Basic usage

- Install the package

```bash
npm i @webapp-suite/ui
```

- Import the component

```jsx
import { Button } from '@webapp-suite/ui';
import '@webapp-suite/ui/dist/index.min.css';

ReactDOM.render(<Button>Name</Button>, mountNode);
```

> Note: @webapp-suite/ui supports ES6 tree shaking by default.

For more information, please refer to [Get Started](https://ui.muwenzi.com/apps/start/usage) in our documentation.

## Local development

### How to run it

```bash
git clone git@github.com:webapp-suite/ui.git
cd ui
yarn
yarn dev
```

Open your browser and visit http://localhost:3003 , see more at [Development Instructions][dev-instructions-url] .

### How to create new component

```bash
yarn create MyComponent
```

Open: http://localhost:3003/apps/components/MyComponent

## Browser support

 | [<img src="https://raw.githubusercontent.com/alrra/browser-logos/master/src/chrome/chrome_48x48.png" alt="Chrome" width="24px" height="24px" />](http://godban.github.io/browsers-support-badges/)</br>Chrome | [<img src="https://raw.githubusercontent.com/alrra/browser-logos/master/src/edge/edge_48x48.png" alt="Edge" width="24px" height="24px" />](http://godban.github.io/browsers-support-badges/)</br>Edge | [<img src="https://raw.githubusercontent.com/alrra/browser-logos/master/src/firefox/firefox_48x48.png" alt="Firefox" width="24px" height="24px" />](http://godban.github.io/browsers-support-badges/)</br>Firefox | [<img src="https://raw.githubusercontent.com/alrra/browser-logos/master/src/safari/safari_48x48.png" alt="Safari" width="24px" height="24px" />](http://godban.github.io/browsers-support-badges/)</br>Safari |
| --- | --- | --- | --- |
| last 2 versions | last 2 versions | last 2 versions | last 2 versions |

---

## Contributing

We welcome all contributions. If you'd like to improve code, please read our [CONTRIBUTING.md][contributing-url] first, check out the [Development Instructions][dev-instructions-url] and feel free to create an [issue][open-issues-url] or submit a [pull request][open-prs-url].

### Maintainers

Welcome to join us if you have interest in this repo, please contact mrgaonju@gmail.com and we can discuss further.

- [Kimi Gao](https://github.com/Kimi-Gao)
- [Robbie Han](https://github.com/USTC-Han)
- [Sherry Shen](https://github.com/Mylittlegirl)

## Changelog

We use [GitHub Releases][github-release-url] to manage our releases, including the changelog between every release. View a complete list of additions, fixes, and changes on the [releases][release-url] page. You can also read [CHANGELOG.md][changelog-url] file.

## License

[![FOSSA Status][fossa-status-image]][fossa-status-url]

All files on the WebApp UI github repository are subjected to the [MIT][license-url] license.

[travis-image]: https://badgen.net/travis/webapp-suite/ui?icon=travis&label=build
[codecov-image]: https://badgen.net/codecov/c/github/webapp-suite/ui/?icon=codecov
[npm-version-image]: https://badgen.net/npm/v/@webapp-suite/ui?icon=npm
[npm-downloads-image]: https://badgen.net/npm/dm/@webapp-suite/ui
[dependencies-image]: https://badgen.net/david/dep/webapp-suite/ui
[open-issues-image]: https://badgen.net/github/open-issues/webapp-suite/ui
[open-prs-image]: https://badgen.net/github/open-prs/webapp-suite/ui
[last-commit-image]: https://badgen.net/github/last-commit/webapp-suite/ui
[contributors-image]: https://badgen.net/github/contributors/webapp-suite/ui
[license-image]: https://badgen.net/npm/license/@webapp-suite/ui
[license-scan-image]: https://app.fossa.io/api/projects/git%2Bgithub.com%2Fwebapp-suite%2Fui.svg?type=shield
[slack-image]: https://badgen.net/badge/icon/slack?icon=slack&label
[fossa-status-image]: https://app.fossa.io/api/projects/git%2Bgithub.com%2Fwebapp-suite%2Fui.svg?type=small

[travis-url]: https://travis-ci.com/webapp-suite/ui
[codecov-url]: https://codecov.io/gh/webapp-suite/ui
[npm-version-url]: https://www.npmjs.com/package/@webapp-suite/ui
[npm-downloads-url]: https://www.npmjs.com/package/@webapp-suite/ui
[dependencies-url]: https://david-dm.org/webapp-suite/ui
[open-issues-url]: https://github.com/webapp-suite/ui/issues
[open-prs-url]: https://github.com/webapp-suite/ui/pulls
[last-commit-url]: https://github.com/webapp-suite/ui/commits/master
[contributors-url]: https://github.com/webapp-suite/ui/graphs/contributors
[license-url]: https://github.com/webapp-suite/ui/blob/master/LICENSE
[license-scan-url]: https://app.fossa.io/projects/git%2Bgithub.com%2Fwebapp-suite%2Fui?ref=badge_shield
[dev-instructions-url]: https://github.com/webapp-suite/ui/wiki/Local-development
[changelog-url]: https://github.com/webapp-suite/ui/blob/master/CHANGELOG.md
[contributing-url]: https://github.com/webapp-suite/ui/blob/master/.github/CONTRIBUTING.md
[slack-url]: https://webapp-suite.slack.com
[fossa-status-url]: https://app.fossa.io/projects/git%2Bgithub.com%2Fwebapp-suite%2Fui?ref=badge_small
[github-release-url]: https://github.com/blog/1547-release-your-software
[release-url]: https://github.com/webapp-suite/ui/releases


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.

## [0.9.0](https://github.com/webapp-suite/ui/compare/v0.8.2...v0.9.0) (2020-05-17)


### Features

* **Button:** add htmlType prop for form submit button ([#278](https://github.com/webapp-suite/ui/issues/278)) ([03fbf72](https://github.com/webapp-suite/ui/commit/03fbf72334a852f6c18135ed7ca4927f019089d9))
* **Nav:** make setting button can be hidden ([#279](https://github.com/webapp-suite/ui/issues/279)) ([e9e121b](https://github.com/webapp-suite/ui/commit/e9e121bded0cc97076d35a7e18590a2358c3587d))


### Bug Fixes

* **button:** prevent button inheriting parent's line-height ([9b48c71](https://github.com/webapp-suite/ui/commit/9b48c718a2ff42740dcd7ed9b6008bd1544e32a7))
* **header:** fix button height issue in safari ([dc9d7d6](https://github.com/webapp-suite/ui/commit/dc9d7d6afe1bf13c850cd1e3173d49226d22f1b1))

### [0.8.2](https://github.com/webapp-suite/ui/compare/v0.8.0...v0.8.2) (2020-05-15)

### Bug Fixes

* **Nav:** render issue in safari when nav collapsed ([#276](https://github.com/webapp-suite/ui/issues/276)) ([244a891](https://github.com/webapp-suite/ui/commit/244a8915fad93e91bcf0be9bb0c37a20f828a22d)), closes [#272](https://github.com/webapp-suite/ui/issues/272)

### [0.8.1](https://github.com/webapp-suite/ui/compare/v0.8.0...v0.8.1) (2020-05-09)

### Bug Fixes

* **Nav:** make Nav can be collapsed after clicked item ([#271](https://github.com/webapp-suite/ui/issues/271)) ([7748b67](https://github.com/webapp-suite/ui/commit/7748b678816955b269772418c85083d07dd95bb3))

## [0.8.0](https://github.com/webapp-suite/ui/compare/v0.7.1...v0.8.0) (2020-04-28)


### Features

* **Header:** support children and polish style ([0cb4c7a](https://github.com/webapp-suite/ui/commit/0cb4c7ae16d2357ce8d41b19367fbc921305baf3))


### Bug Fixes

* **Button:** large button style issue ([687cefa](https://github.com/webapp-suite/ui/commit/687cefaa11a7df1053960ee8fd1e1928c906a3f9))
* **dialog:** make the dialog to the center of window ([fcfc20f](https://github.com/webapp-suite/ui/commit/fcfc20f161d2bd1b5057e41a5cbce27ac769fb5a))
* **docs:** button line-height caused by markdown ([153d289](https://github.com/webapp-suite/ui/commit/153d289a36f95b809e910a6427973699f83a1428))
* **Upload:** onUplading to onUploading ([3ad6871](https://github.com/webapp-suite/ui/commit/3ad68717a7a2caac98a9b10f313ac7476118642f))
* **website:** svg font and style issue ([3092009](https://github.com/webapp-suite/ui/commit/309200978af129eedaf4337cdf088f490b078330))

### [0.7.1](https://github.com/webapp-suite/ui/compare/v0.7.0...v0.7.1) (2020-02-20)


### Bug Fixes

* **deps:** react-custom-scrollbars missing ([c123603](https://github.com/webapp-suite/ui/commit/c123603819214f6910a301b462d97e82d1cff518))

## [0.7.0](https://github.com/webapp-suite/ui/compare/v0.6.0...v0.7.0) (2020-02-20)


### Features

* **input:** add new input component ([#77](https://github.com/webapp-suite/ui/issues/77)) ([653de93](https://github.com/webapp-suite/ui/commit/653de93744bab75d10688260220244c2ffce8b78)), closes [#62](https://github.com/webapp-suite/ui/issues/62)
* **Nav:** make the Nav can collapsed and opened ([#197](https://github.com/webapp-suite/ui/issues/197)) ([98aa328](https://github.com/webapp-suite/ui/commit/98aa328b606a41dac8bcb83f45672a9bf9cae16e))

## [0.6.0](https://github.com/webapp-suite/ui/compare/v0.5.0...v0.6.0) (2019-10-01)


### Features

* **Button:** add loading attribute to button to initialize a spinner ([#76](https://github.com/webapp-suite/ui/issues/76)) ([221b0c5](https://github.com/webapp-suite/ui/commit/221b0c5)), closes [#61](https://github.com/webapp-suite/ui/issues/61)

### Notable changes

* **webpack:**
  * use latest stable webpack@4 ([#56](https://github.com/webapp-suite/ui/pull/56))
  * support webpack tree shaking ([#63](https://github.com/webapp-suite/ui/pull/63))
* **deps:**
  * classlist: update to 2.0.0 ([#69](https://github.com/webapp-suite/ui/pull/69))
  * react-avatar-editor: update to 11.0.7 ([#69](https://github.com/webapp-suite/ui/pull/69))

<a name="0.5.0"></a>
# [0.5.0](https://github.com/webapp-suite/ui/compare/v0.4.1...v0.5.0) (2019-08-11)


### Features

* **Header:** rename HeaderBar to Header and add color prop ([5d5ce5f](https://github.com/webapp-suite/ui/commit/5d5ce5f))
* **Pager:** add Pager component ([#46](https://github.com/webapp-suite/ui/issues/46)) ([74216ec](https://github.com/webapp-suite/ui/commit/74216ec))
* **ToolBar:** add ToolBar for tabs just now ([76849fa](https://github.com/webapp-suite/ui/commit/76849fa))


### BREAKING CHANGES

* **Header:** HeaderBar was obsoleted and use Header instead



<a name="0.4.1"></a>
## [0.4.1](https://github.com/webapp-suite/ui/compare/v0.4.0...v0.4.1) (2019-05-14)


### Bug Fixes

* **Spinner:** text position issue in lg Spinner ([a5f6b57](https://github.com/webapp-suite/ui/commit/a5f6b57))


### BREAKING CHANGES

* **Spinner:** replace `loading` with `isLoading`



<a name="0.4.0"></a>
# [0.4.0](https://github.com/webapp-suite/ui/compare/v0.3.3...v0.4.0) (2019-03-04)


### Features
* **Aside:** add aside and refactor basic dialog by react portal ([#45](https://github.com/webapp-suite/ui/issues/45)) ([fc14984](https://github.com/webapp-suite/ui/commit/eda4016)), closes [#39](https://github.com/webapp-suite/ui/issues/39)
* **dialog/notification:** use colorful buttons and add accept type in dialog ([#41]((https://github.com/webapp-suite/ui/issues/41))) ([eda4016](https://github.com/webapp-suite/ui/commit/eda4016)), closes [#38](https://github.com/webapp-suite/ui/issues/38)
* **Button:** add accept warning and danger button type ([#40](https://github.com/webapp-suite/ui/issues/40)) ([fc14984](https://github.com/webapp-suite/ui/commit/fc14984)), closes [#37](https://github.com/webapp-suite/ui/issues/37)


### BREAKING CHANGES


site/server.js
const express = require('express')
const path = require('path')
const bodyParser = require('body-parser')
const multiparty = require('multiparty')
const compression = require('compression')
// const domain = require('domain')
const fs = require('fs')

const app = express()
app.use(compression())
app.use(bodyParser.json())
app.use(
    bodyParser.urlencoded({
        extended: true
    })
)
app.use(express.static(path.join(__dirname, './')))

// Error handle
// app.use(function (req, res, next) {
//   const reqDomain = domain.create()
//   reqDomain.on('error', function (err) {
//     res.send(500, '<pre>' + err.stack + '</pre>')
//   })
//   reqDomain.run(next)
// })

if (process.env.NODE_ENV === 'development') {
    const webpack = require('webpack')
    const webpackDevMiddleware = require('webpack-dev-middleware')
    const WebpackConfig = require('./webpack.config')
    app.use(
        webpackDevMiddleware(
            webpack({
                ...WebpackConfig,
                mode: 'development'
            }),
            {
                publicPath: WebpackConfig.output.publicPath,
                stats: {
                    colors: true,
                    profile: true
                },
                // you can remove this line for more log about webpack
                // note: https://github.com/webpack/webpack-dev-middleware#loglevel
                logLevel: 'silent'
            }
        )
    )
}

app.post('/api/form', function (req, res) {
    res.json({
        code: 200,
        data: {}
    })
})

app.delete('/user/test', function (req, res) {
    res.json({
        code: 200,
        data: {}
    })
})

function clone (obj) {
    let o
    if (typeof obj === 'object') {
        if (obj === null) {
            o = null
        } else {
            if (obj instanceof Array) {
                o = []
                for (let i = 0, len = obj.length; i < len; i++) {
                    o.push(clone(obj[i]))
                }
            } else {
                o = {}
                for (let j in obj) {
                    o[j] = clone(obj[j])
                }
            }
        }
    } else {
        o = obj
    }
    return o
}

function getTableDate (page, size) {
    const data = [
        {
            id: 1,
            name: '张三',
            age: 28,
            gender: 'male',
            country: '中国',
            area: '北京',
            regdate: '2016-03-01'
        },
        {
            id: 2,
            name: '李四',
            age: 25,
            gender: 'female',
            country: '中国',
            area: '杭州',
            regdate: '2016-04-11'
        },
        {
            id: 3,
            name: '王五',
            age: 43,
            gender: 'male',
            country: '中国',
            area: '沈阳',
            regdate: '2016-05-06'
        },
        {
            id: 4,
            name: '赵某某',
            age: 30,
            gender: 'female',
            country: '中国',
            area: '上海',
            regdate: '2016-03-09'
        },
        {
            id: 5,
            name: '钱某某',
            age: 39,
            gender: 'male',
            country: '中国',
            area: '深圳',
            regdate: '2015-11-11'
        },
        {
            id: 6,
            name: '孙某某',
            age: 50,
            gender: 'male',
            country: '中国',
            area: '石家庄',
            regdate: '2016-06-01'
        },
        {
            id: 7,
            name: '周某某',
            age: 21,
            gender: 'female',
            country: '中国',
            area: '西安',
            regdate: '2016-08-13'
        },
        {
            id: 8,
            name: '吴某某',
            age: 19,
            gender: 'female',
            country: '中国',
            area: '天津',
            regdate: '2016-02-22'
        },
        {
            id: 9,
            name: '郑某某',
            age: 51,
            gender: 'male',
            country: '中国',
            area: '武汉',
            regdate: '2016-01-18'
        },
        {
            id: 10,
            name: '冯某某',
            age: 24,
            gender: 'male',
            country: '中国',
            area: '广州',
            regdate: '2016-09-20'
        }
    ]

    if (page <= 1) {
        return data
    } else {
        const index = page * size
        const newData = []
        for (let i = 0; i < size && i < 10; i++) {
            const item = clone(data[i])
            item.id = item.id + index
            item.name += item.id
            newData.push(item)
        }
        return newData
    }
}

app.get('/api/table', function (req, res) {
    const currentPage = req.query.currentPage || 1
    const pageSize = req.query.pageSize || 10
    const data = getTableDate(currentPage, pageSize)

    res.json({
        code: 200,
        totalList: data,
        currentPage: 1,
        totalPageNum: 300
    })
})

app.post('/upload.do', function (req, res) {
    const form = new multiparty.Form({ uploadDir: './upload' })
    form.on('error', function (err) {
        console.log('Error parsing form: ' + err.stack)
    })
    form.parse(req, function (err, fields, files) {
        const filesTmp = JSON.stringify(files, null, 2)
        try {
            if (err) {
                console.log('parse error: ' + err)
                res.send('写文件操作失败。')
            } else {
                res.send('文件上传成功')
                console.log('parse files: ' + filesTmp)
                const fileNameArr = Object.keys(files)
                const firstFilename = fileNameArr[0]
                const fileDataArr = files[firstFilename]
                const fileData = fileDataArr[0]
                const uploadedPath = fileData.path
                const dstPath = './upload/' + fileData.originalFilename
                fs.rename(uploadedPath, dstPath, function (err) {
                    if (err) {
                        console.log('重命名文件错误：' + err)
                    } else {
                        console.log('重命名文件成功。')
                    }
                })
            }
        } catch (e) {}
    })
    res.json({
        code: 200,
        data: {
            id: new Date().getTime()
        }
    })
})

app.get('*', function (req, res) {
    res.sendFile(path.join(__dirname, 'index.html'))
})

const port = process.env.PORT || 3003

app.listen(port, function () {
    console.log('Server listening on http://localhost:' + port + ', Ctrl+C to stop')
})
